Testscripts/Linux/validate-da.sh

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env bash
+#######################################################################
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# validate-da.sh
+# Description:
+#    Validate the installation and uninstallation of the Dependency Agent.
+#    Validate the enable and disable of the Dependency Agent.
+#######################################################################
+
+set -e
+set -x
+
+# Source utils.sh
+. utils.sh || {
+    echo "ERROR: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+
+# Get distro information
+GetDistro
+
+dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+supported_distro_list=(redhat centos suse debian ubuntu)
+
+readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+
+function test_case_cleanup(){
+    /opt/microsoft/dependency-agent/uninstall
+
+    for dir in "${dir_list[@]}"
+    do
+        rm -rf $dir
+    done
+
+    kill $watch_pid
+
+    LogMsg "Finished cleaning up test cases"
+}
+
+# Calls the cleanup function when interupted or exited
+trap test_case_cleanup SIGINT EXIT
+
+function verify_directory() {
+    failed_assertions=0
+    for dir in "${dir_list[@]}"
+    do
+        [ -d $dir ] && LogErr "Directory $dir exists." && SetTestStateFailed && failed_assertions=$[[failed_assertions+1]]
+    done
+
+    if ["$failed_assertions" -gt 0]; then
+        LogErr "Some/All directories exists"
+        SetTestStateFailed
+        exit 0
+    fi
+}
+
+function verify_distro() {
+    LogMsg "Current distro: $DISTRO"
+    for distro in "${supported_distro_list[@]}"
+    do
+        if [[ "$distro"* == $DISTRO ]]; then
+        LogMsg "Supported Distro family"
+        break
+    else
+        LogErr "Unsupported Distro family"
+        SetTestStateSkipped
+        exit 0
+    done
+}
+
+function verify_da_pid() {
+    [ ! -f $da_pid_file ] && LogErr "$da_pid_file does not exist." && SetTestStateFailed && exit 0
+
+    if x=$(sed -n '/^[1-9][0-9]*$/p;q' $da_pid_file); then
+        strings /proc/$x/cmdline | grep "microsoft-dependency-agent-manager" && LogMsg "PID matched" || (LogErr "PID not matched" && SetTestStateFailed && exit 0)
```

Comment:
Suggestions:
- Use fgrep (or grep -F) since the pattern is not a regex.
- Use the `-q` flag since we aren't interested in the output of grep, only the exit status.

Reply:
Done